Source/lua/lua.cpp

```diff
@@ -152,12 +159,58 @@ void Sol2DebugPrintSection(const std::string &message, lua_State *state)
 	LogDebug("-- {} -- [ {} ]", message, sol::detail::debug::dump_types(state));
 }
 
+sol::environment CreateLuaSandbox()
+{
+	sol::state &lua = CurrentLuaState->sol;
+	sol::environment sandbox(CurrentLuaState->sol, sol::create);
+
+	// Registering globals
+	sandbox.set(
+	    "print", LuaPrint,
+	    "_DEBUG",
+#ifdef _DEBUG
+	    true,
+#else
+	    false,
+#endif
+	    "_VERSION", LUA_VERSION);
+
+	// Register safe built-in globals.
+	for (const std::string_view global : {
+	         // DevilutionX
+	         "Events",
+	         // Built-ins:
+	         "assert", "warn", "error", "ipairs", "next", "pairs", "pcall",
+	         "select", "tonumber", "tostring", "type", "xpcall",
+	         "rawequal", "rawget", "rawset", "setmetatable",
+	// Built-in packages:
+
+#ifdef _DEBUG
+	         "debug",
+#endif
+	         "base", "coroutine", "table", "string", "math", "utf8" }) {
+		const sol::object obj = lua[global];
+		if (obj.get_type() == sol::type::nil) {

```

Comment:
Apparently Mac doesn't like this :thinking: 

error: no member named 'nil' in 'sol::type'

Missing a header?

Reply:
Fixed! Using `lua_nil` instead for better Mac compat